bin/locale_update.py

```diff
@@ -72,12 +72,27 @@ def create_locale(src_dir):
         jar_mn = jar_file.read()
 
         # Replace multiple locale registrations with new locales.
-        jar_locales = ['% locale loop {0} %locale/{0}/'.format(x) for x in locale_list]
+        # The jar.mn preprocessor can't cope with '-' so we add some defines
+        # in so it can handle '_' instead.
+        dashLocales = []
+        for locale in locale_list:
+            if "-" in locale:
+                dashLocales.append('#define {0} {1}'.format(locale.replace("-", "_"), locale))
+
         new_content = re.sub(
-            '(% locale loop .+\n)+',
-            '\n'.join(jar_locales) + '\n',
+            '(#define .+\n)+',
```

Comment:
This could be dangerous if we needed to `#define` other stuff, but `#define LOCALE_zh_TW zh-TW` seems a bit silly. (Where before `% locale loop` had a bit more "context" that this was something specific to locales.) This should be fine for now.